rfcs/80-curve-trait.md

```diff
@@ -0,0 +1,611 @@
+Feature Name: `curve-trait`
+
+## Summary
+
+This RFC introduces a general trait API, `Curve<T>`, for shared functionality of curves within Bevy's
+ecosystem. This encompasses both *abstract* curves, such as those produced by splines (and hence defined
+by equations), as well as *concrete* curves that are defined by sampling and interpolation (e.g. 
+animation keyframes), among others. 
+
+The purpose is to provide a common baseline of useful functionality that unites these disparate
+implementations, making curves of many kinds pleasant to work with for authors of plugins and games.
+Common uses of curves in games include:
+* Describing animations as transformations over time
+* Describing movement of game entities
+* Describing camera moves
+* Describing easing and attenuation with time and/or distance
+* Describing geometry (e.g. of roads, surfaces of extrusion, etc.)
+* Describing paths of particles and effects
+* Defining experience and drop rate curves
+
+## Motivation
+
+Curves tend to have quite disparate underlying implementations at the level of data. As a result,
+they are prone to causing ecosystem fragmentation, as authors of various plugins (including internal ones)
+are liable to create the curve abstractions that suit their own particular needs.
+
+By providing a common baseline in `Curve<T>`, we can ensure a significant degree of portability between these
+disparate areas and, ideally, allow for authors to reuse the work of others rather than reinventing the
+wheel every time they want to write libraries that involve generating or working with curves. 
+
+Furthermore, something like this is also a prerequisite to bringing more complex geometric operations
+(some of them with broad implications) into `bevy_math` itself. For instance, this lays the natural bedwork
```

Comment:
```suggestion
(some of them with broad implications) into `bevy_math` itself. For instance, this lays the natural foundation
```

"bedwork" is not an idiom in English :)

Reply:
This feels like I engaged in an AI hallucination.

Reply:
looks like a frankenstein portmanteau of "bedrock" and "groundwork" 😛